test cases/python/7 install path/meson.build

```diff
@@ -9,4 +9,8 @@ py = import('python').find_installation()
 py.install_sources('test.py')
 py.install_sources('test.py', pure: false)
 
+py_plat = import('python').find_installation(pure: false)
+py_plat.install_sources('test.py', subdir: 'kw')
+py_plat.install_sources('test.py', pure: true, subdir: 'kwrevert')
```

Comment:
May be good to add a test file using `get_install_dir` as well?

Reply:
@eli-schwartz I just discovered that this PR in its current state actually breaks `get_install_dir(pure: false)`. To reproduce:
```
$ # use current SciPy `main`
$ meson setup build --prefix=$PWD/build-install
$ grep purelib build/meson-info/intro-install_plan.json | wc -l
# will give 0 with meson master, and 1 with this PR
```

Reply:
Yeah, that was dumb of me and happened because I didn't fully revert my first attempt.